discovery-provider/src/tasks/index_aggregate_tips.py

```diff
@@ -49,6 +94,34 @@ def _update_aggregate_tips(session: Session):
         "slot",
         max_slot,
     )
+    ranks_after = _get_ranks(session, prev_slot, max_slot)
+    grouped_ranks_before = {
```

Comment:
this is interesting pattern, is this like object comprehension or smthg?

Reply:
dict comprehension! it's pretty handy